App consistently crashes when using "Remove collection revision"

# Bug Report

## Summary
The app crashes when a revision is removed from the collection download page. 

## Steps to reproduce
1. Build from 71b7acaa94b8420b2642883ac2ebb8c16e3bd9f2 
3. Install https://next.nexusmods.com/stardewvalley/collections/tjvl0j
4. Remove it from the collection download page.
5. App crashes.## What is the expected behaviour?
The app does not crash.

## Other information

https://github.com/user-attachments/assets/62d3c16a-1153-440f-a0ff-1de2c92da331

Crash Log
```
00:30:18.434 [ERROR] (NexusMods.App.Program) Encountered an unobserved task exception in the Task Scheduler from sender of type `System.Threading.Tasks.Task`1[System.Object]`: `System.Threading.Tasks.Task`1[System.Object]`|System.InvalidOperationException: Sequence contains no elements
   at System.Linq.ThrowHelper.ThrowNoElementsException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at NexusMods.App.UI.Pages.LibraryPage.Collections.CollectionCardViewModel..ctor(CollectionDownloader collectionDownloader, IResourceLoader`2 tileImagePipeline, IResourceLoader`2 userAvatarPipeline, IWindowManager windowManager, WorkspaceId workspaceId, IConnection connection, RevisionId revision, LoadoutId targetLoadout) in C:\Projects\NexusMods.App\src\NexusMods.App.UI\Pages\LibraryPage\Collections\CollectionCardViewModel.cs:line 37
   at NexusMods.App.UI.Pages.LibraryPage.LibraryViewModel.<>c__DisplayClass47_0.<.ctor>b__15(ReadOnly revision) in C:\Projects\NexusMods.App\src\NexusMods.App.UI\Pages\LibraryPage\LibraryViewModel.cs:line 197
   at DynamicData.ObservableCacheEx.<>c__DisplayClass211_0`3.<Transform>b__0(TSource current, Optional`1 _, TKey _) in /_/src/DynamicData/Cache/ObservableCacheEx.cs:line 4608
   at DynamicData.Cache.Internal.Transform`3.<RunImpl>b__6_0(ChangeAwareCache`2 cache, IChangeSet`2 changes) in /_/src/DynamicData/Cache/Internal/Transform.cs:line 46
   at System.Reactive.Linq.ObservableImpl.Scan`2._.OnNext(TSource value)
--- End of stack trace from previous location ---
   at System.Reactive.PlatformServices.ExceptionServicesImpl.Rethrow(Exception exception)
   at System.Reactive.ExceptionHelpers.Throw(Exception exception)
   at System.Reactive.Stubs.<>c.<.cctor>b__2_1(Exception ex)
   at System.Reactive.AnonymousObserver`1.OnErrorCore(Exception error)
   at System.Reactive.ObserverBase`1.OnError(Exception error)
   at System.Reactive.AutoDetachObserver`1.OnErrorCore(Exception exception)
   at System.Reactive.ObserverBase`1.OnError(Exception error)
   at System.Reactive.Sink`1.ForwardOnError(Exception error)
   at System.Reactive.Sink`2.OnError(Exception error)
   at System.Reactive.Sink`1.ForwardOnError(Exception error)
   at System.Reactive.Concurrency.Synchronize`1._.OnError(Exception error)
   at System.Reactive.Sink`1.ForwardOnError(Exception error)
   at System.Reactive.Sink`2.OnError(Exception error)
   at System.Reactive.Sink`1.ForwardOnError(Exception error)
   at System.Reactive.Sink`2.OnError(Exception error)
   at System.Reactive.Sink`1.ForwardOnError(Exception error)
   at System.Reactive.Sink`2.OnError(Exception error)
   at System.Reactive.Sink`1.ForwardOnError(Exception error)
   at System.Reactive.Linq.ObservableImpl.Scan`2._.OnNext(TSource value)
   at System.Reactive.ObserveOnObserverNew`1.DrainShortRunning(IScheduler recursiveScheduler)
   at Avalonia.ReactiveUI.AvaloniaScheduler.<>c__DisplayClass4_1`1.<Schedule>b__1()
   at Avalonia.Threading.DispatcherOperation.InvokeCore()
   at Avalonia.Threading.DispatcherOperation.Execute()
   at Avalonia.Threading.Dispatcher.ExecuteJob(DispatcherOperation job)
   at Avalonia.Threading.Dispatcher.ExecuteJobsCore(Boolean fromExplicitBackgroundProcessingCallback)
   at Avalonia.Threading.Dispatcher.Signaled()
   at Avalonia.Win32.Win32Platform.WndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam)
   at Avalonia.Win32.Interop.UnmanagedMethods.DispatchMessage(MSG& lpmsg)
   at Avalonia.Win32.Win32DispatcherImpl.RunLoop(CancellationToken cancellationToken)
   at Avalonia.Threading.DispatcherFrame.Run(IControlledDispatcherImpl impl)
   at Avalonia.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at Avalonia.Threading.Dispatcher.MainLoop(CancellationToken cancellationToken)
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.StartCore(String[] args)
   at Avalonia.Controls.ApplicationLifetimes.ClassicDesktopStyleApplicationLifetime.Start(String[] args)
   at Avalonia.ClassicDesktopStyleApplicationLifetimeExtensions.StartWithClassicDesktopLifetime(AppBuilder builder, String[] args, Action`1 lifetimeBuilder)
   at NexusMods.App.Startup.Main(IServiceProvider provider, String[] args) in C:\Projects\NexusMods.App\src\NexusMods.App\Startup.cs:line 57
```

[nexusmods.app.main.current.log](https://github.com/user-attachments/files/19006940/nexusmods.app.main.current.log)
[nexusmods.app.slim.current.log](https://github.com/user-attachments/files/19006941/nexusmods.app.slim.current.log)

--- a/src/NexusMods.App.UI/Pages/LibraryPage/Collections/CollectionCardViewModel.cs
+++ b/src/NexusMods.App.UI/Pages/LibraryPage/Collections/CollectionCardViewModel.cs
@@ -30,11 +30,10 @@ public class CollectionCardViewModel : AViewModel<ICollectionCardViewModel>, ICo
         IResourceLoader<EntityId, Bitmap> userAvatarPipeline,
         IWindowManager windowManager,
         WorkspaceId workspaceId,
-        IConnection connection,
-        RevisionId revision,
+        CollectionRevisionMetadata.ReadOnly revision,
         LoadoutId targetLoadout)
     {
-        _revision = CollectionRevisionMetadata.FindByRevisionId(connection.Db, revision).First();
+        _revision = revision;
         _collection = _revision.Collection;
 
         var workspaceController = windowManager.ActiveWorkspaceController;

--- a/src/NexusMods.App.UI/Pages/LibraryPage/LibraryViewModel.cs
+++ b/src/NexusMods.App.UI/Pages/LibraryPage/LibraryViewModel.cs
@@ -200,8 +200,7 @@ public class LibraryViewModel : APageViewModel<ILibraryViewModel>, ILibraryViewM
                     userAvatarPipeline: userAvatarPipeline,
                     windowManager: WindowManager,
                     workspaceId: WorkspaceId,
-                    connection: _connection,
-                    revision: revision.RevisionId,
+                    revision: revision,
                     targetLoadout: _loadout)
                 )
                 .Bind(out _collections)

